### ISSUE TITLE
Web Browsers Under Ubuntu Cannot connect to h2o when "cipher-preference: server" option is added.

### ISSUE THREAD

**Issue by futursolo** [h2o/h2o#342]
Both Chrome and Firefox have this issue under Ubuntu.

I suppose this issue is given rise by OpenSSL.

when I set "cipher-preference: server" in my h2o configuration file, Chrome and Firefox under Ubuntu cannot connect to my website, but those for Windows work normally.

**Comment by kazuho** [comment]
Sorry to be late, I tried but could not reproduce the issue.

My environment is Ubuntu 14.04 + Firefox 38 (Mozilla Firefox for Ubuntu canonical - 1.0).  I tried accessing https://h2o.examp1e.net (which runs H2O), and the browser successfully displayed the web page.

Would you mind accessing https://h2o.examp1e.net and report back what happens?  If you can see the page, then it is likely an issue with how you build H2O.

**Comment by futursolo** [comment]
My configuration file:
```
hosts:
    "www.example.com:443":
        header.set: "strict-transport-security: max-age=39420000; includesubdomains; preload"
        listen:
            port: 443
            ssl:
                certificate-file: /etc/h2o/server.crt.pem
                key-file: /etc/h2o/server.key.pem
                cipher-preference: server
                dh-file: /etc/h2o/dh.pem
                cipher-suite: {any cipher suite, such as: https://wiki.mozilla.org/Security/Server_Side_TLS}
        paths:
            /:
                proxy.reverse.url: http://127.0.0.1:1741/
                proxy.preserve-host: ON
                proxy.timeout.io: 30000
```
The configuartion above will occur "ERR_SPDY_INADEQUATE_TRANSPORT_SECURITY" in Chrome, and in Firefox, blank page. If I turn to configuration file below, both Chrome and Firefox work normally.
```
hosts:
    "www.example.com:443":
        header.set: "strict-transport-security: max-age=39420000; includesubdomains; preload"
        listen:
            port: 443
            ssl:
                certificate-file: /etc/h2o/server.crt.pem
                key-file: /etc/h2o/server.key.pem
                # cipher-preference: server # Just comment this line
                dh-file: /etc/h2o/dh.pem
                cipher-suite: {any cipher suite}
        paths:
            /:
                proxy.reverse.url: http://127.0.0.1:1741/
                proxy.preserve-host: ON
                proxy.timeout.io: 30000
```

Additionally, I am able to access the https://h2o.examp1e.net/.

**Comment by retsaot** [comment]
@futursolo Do you have "ECDHE-ECDSA-AES128-GCM-SHA256", "ECDHE-RSA-AES128-GCM-SHA256" and "DHE-RSA-AES128-GCM-SHA256" ciphers enabled? Without them, I also get "ERR_SPDY_INADEQUATE_TRANSPORT_SECURITY".

**Comment by futursolo** [comment]
@retsaot It seems that this issue has been resolved on h2o 1.3.1 and Ubuntu 14.04 LTS version.
